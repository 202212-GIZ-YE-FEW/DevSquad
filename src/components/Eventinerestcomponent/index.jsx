import Checkboxcomponent from "../Checkboxcomponent";
const Eventinerestcomponent = () => {
    const titles = [
        "All",
        "No Poverty",
        "Zero Hunger",
        "Good Health And Well-Being",
        "Quality Education",
        "Gender Equality",
        "Clean Water And Sanitation",
        "Affordable And Clean Energy",
        "Decent Work And Economic Growth",
        "Industry Innovation And Infrastructure",
        "Reduced Inequalities",
        "Sustainable Cities And Communities",
        "Responsible Consumption And Production",
        "Climate Action",
        "Life Below Water",
        "Life On Land",
        "Peace And Justice And Strong Institutions",
    ];

    return (
        <>
            {/* desktop design */}
            <div className='md:w-64 sm:w-56 w-full bg-white sm:h-full h-[50%] overflow-auto scrollbar fixed sm:static sm:z-0 z-50 top-1/2 left-1/2 sm:transform-none transform -translate-x-1/2 shadow-inner sm:shadow-none shadow-gray-700 rounded-lg'>
                <div className='grid grid-cols-1 gap-2 '>
                    <p className='font-medium font-Rubik underline text-center sm:block hidden'>
                        Pick Your Interest
                    </p>
                    <p className='font-medium font-Rubik text-center block sm:hidden text-lg my-2'>
                        Change Interset
                    </p>
                    {titles &&
                        titles.map((title) => {
                            return (
                                <Checkboxcomponent
                                    key={title}
                                    title={title}
                                    afterChecked='flex items-center justify-center text-center sm:border-2 sm:border-black border-0 sm:bg-secondry-orange bg-white md:p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs'
                                    beforeChecked='checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 md:p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs'
                                    view='sm:hidden'
                                    block='sm:block'
                                    flex='flex'
                                    intrestMargin='ml-2'
                                    checkMargin='sm:m-0 m-6'
                                />
                            );
                        })}
                </div>
            </div>
        </>
    );
};

export default Eventinerestcomponent;
