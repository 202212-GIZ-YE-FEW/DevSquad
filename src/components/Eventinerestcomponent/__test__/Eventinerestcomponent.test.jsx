import renderer from "react-test-renderer";

import Eventinerestcomponent from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<Eventinerestcomponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
