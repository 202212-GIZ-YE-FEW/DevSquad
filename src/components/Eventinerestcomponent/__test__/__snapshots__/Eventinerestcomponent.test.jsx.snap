// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="sm:mx-32 mx-10 sm:w-80 sm:p-5 sm:block sm:static sm:z-0	fixed z-50 top-0 left-0 sm:h-full h-96 bg-opacity-50 sm:opacity-100 transition-opacity  opacity-0 pointer-events-none"
  >
    <div
      className="fixed sm:static sm:z-0 z-50 top-1/2 left-1/2 sm:transform-none transform -translate-x-1/2 -translate-y-1/2 bg-white rounded-md p-4  w-full sm:w-80 h-80 sm:h-full overflow-y-scroll sm:overflow-auto scrollbar "
    >
      <div
        className="grid grid-cols-1 gap-2 "
      >
        <p
          className="font-medium font-Rubik underline text-center sm:block hidden"
        >
          Pick Your Interest
        </p>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              All
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              No Poverty
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Zero Hunger
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Good Health And Well-Being
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Quality Education
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Gender Equality
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Clean Water And Sanitation
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Affordable And Clean Energy
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Decent Work And Economic Growth
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Industry Innovation And Infrastructure
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Reduced Inequalities
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Sustainable Cities And Communities
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Responsible Consumption And Production
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Climate Action
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Life Below Water
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Life On Land
            </span>
          </label>
        </div>
        <div>
          <label
            className="flex sm:block"
          >
            <input
              className="sm:hidden"
              type="checkbox"
            />
            <span
              className="checked flex items-center justify-center text-center  sm:border-2 sm:border-black border-0 p-3 rounded h-16  font-Rubik font-medium sm:text-base text-xs"
            >
              Peace And Justice And Strong Institutions
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>,
  <button
    className="flex items-center  p-2 border border-black rounded mt-4 md:hidden"
    onClick={[Function]}
  >
    <p
      className="px-2 font-Rubik"
    >
      Change Interest
    </p>
    <svg
      fill="none"
      height="8"
      viewBox="0 0 13 8"
      width="13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.4704 0.705392C11.6865 0.465228 12.0564 0.445759 12.2965 0.661906C12.5366 0.878053 12.5561 1.24797 12.34 1.48813L7.07538 7.3385C6.84301 7.59672 6.43816 7.59672 6.20579 7.3385L0.941182 1.48813C0.725065 1.24797 0.744531 0.878053 0.984662 0.661906C1.22479 0.445759 1.59465 0.465228 1.81077 0.705392L6.64058 6.07259L11.4704 0.705392Z"
        fill="#1A1A1A"
      />
    </svg>
  </button>,
]
`;
